### PR TITLE
fix console hanging, always flush log, vstsagent_trace env.

### PR DESCRIPTION
**Pull request by TingluoHuang**

### CODE REVIEW

File: src/Agent.Listener/Program.cs [hunk 1]
```diff
@@ -57,47 +57,44 @@ public static int Main(string[] args)
                         return Constants.Agent.ReturnCode.TerminatedError;
                 }
 
-                using (var term = context.GetService<ITerminal>())
```
Comment (TingluoHuang) [inline src/Agent.Listener/Program.cs]: we should not dispose the terminal.

Comment (ericsciple) [inline src/Agent.Listener/Program.cs]: It looks like that's the only thing that changed in this block, is that correct?

Comment (TingluoHuang) [inline src/Agent.Listener/Program.cs]: yes

**Comment by ericsciple** [comment]
looks good

**Pull request merged**